### PR TITLE
[MIG] account_multicompany_code: Migrate to v15

### PR DESCRIPTION
**Pull request by JesusValdez96**
- Fix black, isort and prettier lints
- Fix pylints in res_company model (use Python 3 style super() without arguments)

### CODE REVIEW

**Comment by JesusValdez96** [comment]
@hugho-ad Please review this.

**Review by luisg123v — CHANGES_REQUESTED** [review]
We accorded by mail when this module was migrated to v14 that we will depend on the module `res_company_code`, because its constraint is more efficient (it's an SQL constraint instead of a Python one) and the code is considered also on the company switcher. That module is [here](https://github.com/OCA/multi-company/tree/14.0/res_company_code)

**Comment by hugho-ad** [comment]
@luisg123v 

why not to use only the oca one?

the mig to v14 was performed by you
https://github.com/Vauxoo/addons-vauxoo/pull/1458

**Comment by hugho-ad** [comment]
ya lei los hilos completos sobre este modulo 
 creo que @nhomar no espera esta dependencia del modulo de la oca.

En mi opinión deberiamos manterner nuestro modulo sin depender  de la oca

**Comment by luisg123v** [comment]
No, what @nhomar said was to don't replace this module with the OCA's one, because this module also adds other feature.

This does two things:
1. Add a code to the company and ensure it's unique
2. Use that code in accounts

The 1st point is covered by the OCA module on a more complete and efficient manner, so I think it should be used, and only the 2nd point should be covered from our side

**Review by hugho-ad — APPROVED** [review]

**Comment by hugho-ad** [comment]
Strategic merge

We will NOT depend on OCA module, this will remain as it's!

File: account_multicompany_code/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "author": "Vauxoo",
+    "category": "Accounting/Accounting",
+    "website": "https://www.vauxoo.com/",
+    "license": "LGPL-3",
+    "depends": [
+        "account",
+    ],
+    "demo": [],
+    "data": [
+        "views/company_view.xml",
```
Comment (luisg123v) [inline account_multicompany_code/__manifest__.py:13]: `res_company_views.xml`

FYI (for next migration) black-related changes goes into a separated commit, to be able to distinguish formatting changes from actual migration changes

File: account_multicompany_code/models/account.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    def name_get(self):
+        result = dict(super().name_get())
+        for record in self:
+            if record.company_id.code:
+                result[record.id] += " (%s)" % record.company_id.code
+        return list(result.items())
+
+
+class AccountJournal(models.Model):
```
Comment (luisg123v) [inline account_multicompany_code/models/account.py:15]: Move to its own file, each model should be placed on a separated file

File: account_multicompany_code/models/res_company.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+from odoo.exceptions import ValidationError
+
+
+class Company(models.Model):
+    _inherit = "res.company"
+
+    code = fields.Char(help="Internal code name of the company")
+
+    @api.constrains("code")
+    def unique_code(self):
```
Comment (luisg123v) [inline account_multicompany_code/models/res_company.py:11]: Move to SQL constraint, for efficiency, as the module `res_company_code` does

File: account_multicompany_code/tests/test_multicompany_code.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+from odoo.tests import TransactionCase
+
+
+class TestMulticompanyCode(TransactionCase):
+    def setUp(self):
```
Comment (luisg123v) [inline account_multicompany_code/tests/test_multicompany_code.py:5]: `setUpClass`

File: account_multicompany_code/views/company_view.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record id="view_company_form" model="ir.ui.view">
+        <field name="name">qf_common.company.form</field>
+        <field name="model">res.company</field>
+        <field name="inherit_id" ref="base.view_company_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='partner_id']" position="before">
+                <field name="code" required="1" />
```
Comment (luisg123v) [inline account_multicompany_code/views/company_view.xml:10]: Missing changes on other views, check module `res_company_code` for more context

**Review by luisg123v — COMMENTED** [review]
Better late than never

**Comment by luisg123v** [comment]
@LuisMzz could you take care of the review I left post migration, please?

**Comment by luisg123v** [comment]
@hugho-ad prefer merging using the rebase method, to don't create a merge commit.

**Pull request merged**